### PR TITLE
[HW-2] Implement my own Array#my_each, Array#my_map, Array#my_select methods

### PR DESCRIPTION
**Pull request by m3xq**
[HW-2] Implement my own Array methods

### CODE REVIEW

File: HW02/vladislav_mitin/hw02.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+module MyEnumerableMethods
+  refine Array do
+    def my_each
+      if block_given?
+        0.upto(length - 1) { |i| yield self[i] }
```
Comment (B0RlS) [inline HW02/vladislav_mitin/hw02.rb:5]: Look at the times loop, maybe you can use it.

Comment (m3xq) [reply HW02/vladislav_mitin/hw02.rb]: Are there still some issues?

Comment (B0RlS) [reply HW02/vladislav_mitin/hw02.rb]: no, everything is ok

File: HW02/vladislav_mitin/my_enumerable_methods.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    end
+
+    def my_map
+      res = []
+
+      if block_given?
+        length.times { |i| res << yield(self[i]) }
+      else
+        res = to_enum :my_map
+      end
```
Comment (B0RlS) [inline HW02/vladislav_mitin/my_enumerable_methods.rb:20]: If you like, you can shorten the code using the ternary conditional operator
Also in the 28-32 line

**Review by vladhilko — APPROVED** [review]

**Review by B0RlS — APPROVED** [review]

**Pull request merged**